### PR TITLE
Add a plugin to inhibit idle(screensaver)/auto suspend

### PR DESCRIPTION
**Pull request by JasonLG1979**

### CODE REVIEW

**Comment by TingPing** [comment]
Should it really be a plugin?

**Comment by JasonLG1979** [comment]
> Should it really be a plugin?

If by that you mean "Is this a useful feature?" I'd say yes. If by that you mean "Why make a built in Gtk function a plugin?" I'd say because that's pretty much where we put most of our user options.

File: pithos/plugins/inhibit_screensaver.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        idle = Gtk.ApplicationInhibitFlags.IDLE
+
+        self._cookie = self._pithos_app.inhibit(
+            self.window,
+            suspend | idle,
+            'Inhibit Screensaver plugin enabled',
+        )
+
+    def _uninhibit(self):
+        if self._cookie and self._cookie != 0:
```
Comment (TingPing) [inline pithos/plugins/inhibit_screensaver.py]: redundant

**Comment by TingPing** [comment]
Well I was just thinking it has no dependencies and seems logical as default behavior but there is always somebody who would probably want to disable it.

File: pithos/plugins/inhibit_screensaver.py (hunk 1 above)
Comment (JasonLG1979) [inline pithos/plugins/inhibit_screensaver.py]: fixed.

**Comment by JasonLG1979** [comment]
> Well I was just thinking it has no dependencies and seems logical as default behavior but there is always somebody who would probably want to disable it.

I'm not sure I'd make it default behavior. I like it better as an option. I kinda see it as a complimenting, almost opposite option to the Screensaver Pause plugin. Which I see as a better default behavior.

**Pull request merged**